app/models/ont/flowcell.rb

```diff
@@ -8,15 +8,82 @@ class Flowcell < ApplicationRecord
     # Run has many of these flowcells up to the maximum number for the instrument.
     belongs_to :run, foreign_key: :ont_run_id, inverse_of: :flowcells
 
-    # We assume one-to-one relationship with pool at the moment.
-    belongs_to :pool, foreign_key: :ont_pool_id, inverse_of: :flowcell
+    # We assume one-to-one relationship with pool. We make it optional here to
+    # customise the validation later.
+    belongs_to :pool, foreign_key: :ont_pool_id, inverse_of: :flowcell, optional: true
 
     delegate :requests, :libraries, to: :pool
 
-    # Validate the position to be a positive integer.
-    validates :position,
-              presence: true,
-              numericality: { greater_than_or_equal_to: 1 }
-    validates :flowcell_id, presence: true
+    # flowcell position validations
+    validates :position, presence: true
+    validates :position, numericality: { only_integer: true }
+    validates :position, numericality: {
+      greater_than_or_equal_to: 1,
+      less_than_or_equal_to: :max_number_of_flowcells,
+      if: :max_number_of_flowcells,
+      message: lambda do |_object, data|
+        "position #{data[:value]} is out of range for the instrument"
+      end
+    }
+    validates :position, uniqueness: {
+      scope: :run,
+      message: lambda do |_object, data|
+        "position #{data[:value]} is duplicated in the same run"
+      end
+    }
+
+    # pool validations
```

Comment:
Not entirely sure what is going on here? What is object.device_id?

Reply:
UI asked to give coordinates for PromethION flowcells, instead of numbers, in error messages. device_id either returns a coordinate such as 1A or a position.

Reply:
fyi Coordinates will be row then column, so A1 not 1A

Reply:
I have replaced the occurrences of device_id with position_display now. 
I have change the places of letter and number for promethION.